addon/components/polaris-button-group.js

```diff
@@ -66,11 +67,13 @@ export default Component.extend({
     this._super(...arguments);
 
     // Wrap each child element that isn't already a group item.
-    this.$()
-      .children()
-      .not('div.Polaris-ButtonGroup__Item')
-      .wrap(
-        '<div data-test-button-group-item class="Polaris-ButtonGroup__Item"></div>'
-      );
+    let wrapper = document.createElement('div');
+
+    wrapper.classList.add('Polaris-ButtonGroup__Item');
+    wrapper.setAttribute('data-test-button-group-item', true);
+
+    wrapChildren(this.element.children, wrapper, function(el) {
```

Comment:
I'm good with this, when I refactor you might see a bit of a name change in some spots since some of the usages aren't only specific to `class` names, we also check for things like `role` in the form layout groups. But either way 👍